### ISSUE TITLE
Github Desktop swap commits between branches

### ISSUE THREAD

**Issue by msPascal** [desktop/desktop#18017]
### The problem

Feature branch changes been transferred to staging when checkout to staging. Commit is seen as a staging commit even is a feature branch with a PR in place and no uncommitted changes on staging 

It happened more than once in last days. 

### Release version

Version 3.3.6 (arm64)

### Operating system

Sonoma Version 14.2.1 (23C71)

### Steps to reproduce the behavior

Create a feature branch 
Pushed a commit change. 
Create a PR to merge to staging and all was looking ok. 
Open Github Desktop 
Checkout to staging using the Branch dropdown. 
Once succeed the commit from the feature branch move to staging
Push 1 commit log is displayed and Pushed button been activated. 
Fetch the staging to be up to date commit is pushed, Pr closed, branched merged and bypass all rights. 

### Screenshots

<img width="1312" alt="Screenshot 2024-01-15 at 14 56 49" src="https://github.com/desktop/desktop/assets/1441380/e7bdc758-440c-41cf-bc86-676decc756ff">
<img width="846" alt="Screenshot 2024-01-15 at 14 59 03" src="https://github.com/desktop/desktop/assets/1441380/cd8293e6-fffa-4a7c-96de-25d9495a9507">
![Untitled12](https://github.com/desktop/desktop/assets/1441380/fc86b9f8-dae2-4dd2-9477-99a83a58cf8a)

**Comment by steveward** [comment]
Thanks for the report @pia90. I'm not sure I understand the issue. Are you saying uncommitted changes have been moved from one branch to another branch, or an existing commit? An existing commit won't be carried over unless you are cherry-picking a commit or performing a merge. Is there a chance you are doing either of those by accident? 

If you open the repository in the command line and run `git reflog` that will also give a detailed history of what commands were run in the repository, which could be helpful in diagnosing this issue.

**Comment by msPascal** [comment]
Hello @steveward. Thank you coming back. The commit carried are committed and I didn't use cherry pick. I checkout from my feature branch to staging and that commit was in moved to staging . It happened few times now. 
Steps.
- I did a feature branch. I made changes and I commit the changes in one commit. I pushed and created a PR for feature branch to staging 
- Went to desktop app and I checkout to staging. Feature commit was in staging and push button activated. 
- In other case. Push pull was activated if my local staging wasn't up-to-date.
